### ISSUE TITLE
Fast R-CNN

### ISSUE THREAD

**Issue by sooooner** [sooooner/papers#33]
# Fast R-CNN[[link](https://arxiv.org/abs/1504.08083)]

### R-CNN and SPPnet
R-CNN의 단점
1. Training is a multi-stage pipeline.
  + selective search
  + SVM classification
  + bounding box regression
2. Training is expensive in space and time
  + SVM 너무 느리다
  + 2000개의 proposal을 disk에 올린다? 개비쌈!
3. Object detection is slow
  + test time에서도 느림

R-CNN은 각 입력에 대해 2000개의 영역을 제시하고 각각 conv layer를 통과시켜(공유하지 않음) 비효율적임
SPP Net은 이를 어느 정도 해결함 - 전체 image에서 Conv한뒤 feature map에서 proposal을 추출
결과로 traning은 3배, cnfhsdms 10~100배 빨라짐
하지만 SPP Net도 multi-stage와 disk 문제를 가지고 있음

### Contributions
R-CNN과 SPPNet의 단점을 고친 fast RCNN을 소개한다.
1. 좋은 성능
2. single stage using multi task loss
3. end-to-end
4. disk caching X

## Fast R-CNN architecture and training

![image](https://user-images.githubusercontent.com/31437443/112925776-5b960600-914d-11eb-8f77-8c9a412c0221.png)

### The RoI pooling layer
RCNN의 selective search가 Conv로 바뀜
RCNN에서 resizing이 RoI pooling이 됨

![image](https://user-images.githubusercontent.com/31437443/112925514-f2ae8e00-914c-11eb-9397-3e9ced8692df.png)

### Initializing from pre-trained networks

image net에서 훈련된 model의 conv layers를 사용함(VGG)으로써 feature extractor의 성능 향상
SVM이 softmax FC layer로 바뀜

#### Multi-task loss

softmax에서 나온 각 class에 대한 확률값, regressor에서 나온 값으로 같이 학습하는 Multi-Task Learning을 사용함
-> 성능을 전반적으로 향상 

![image](https://user-images.githubusercontent.com/31437443/112925719-402afb00-914d-11eb-9270-81e34d288e81.png)

1. cls Loss : log loss
2. loc Loss : smooth L1 loss
3. lambda : 두 task의 balance 조절 변수(논문에선 lambda = 1)

#### Mini-batch sampling
각 SGD mini batch는 N = 2 images로 구성됨
각 이미지에서 RoIs는 64개로 R = 64 * 2 = 128
ground-truth와 IoU가 0.5이상인 값을 사용함

#### Back-propagation through RoI pooling layer
xi이 roi pooling layer로 i-th activation input이라고 하자. 
yrj는 r-th roi의 j-th output이라고 하자.

그럼 roi pooling layer는 yrj = x i*(r, j)로 계산된다 여기서 i*(r, j) = argmax xi' 

미분은 아래와 같이 된다.
![image](https://user-images.githubusercontent.com/31437443/112927490-08718280-9150-11eb-9aa4-8ce625f4ff6d.png)

## Main results

1. State-of-the-art mAP on VOC07, 2010, and 2012
2. Fast training and testing compared to R-CNN, SPPnet
3. Fine-tuning conv layers in VGG16 improves mAP
